### ISSUE TITLE
[React-Native][5.1.X] addon-ondevice-notes crash on initial story selection

### ISSUE THREAD

**Issue by kenny-inthirath-pear** [storybookjs/storybook#7606]
**Describe the bug**
Using the Notes add-on for the INITIAL Story that is loaded crashes the app and throws a JS error

Any Story selected after the initial one WILL NOT crash

**To Reproduce**
1. Go to 'ADDONS' on the INITIAL Story loaded
2. Click on 'NOTES'
3. See error `TypeError: Cannot read property of 'selection' of null`

**Expected behavior**
Should see the text passed in (or auto-generated notes)

**Screenshots**
(with bug and workaround)
![StorybookNotes](https://user-images.githubusercontent.com/53273174/62090131-9630f800-b220-11e9-8f3a-bc6fea34d53e.gif)

**Code snippets**
```js
// rn-addons.js
import '@storybook/addon-ondevice-notes/register';

...

// someStory.js
simpleComponentsStory.add('someName', SomeComponent, { notes: 'some notes'})
```

**System:**
```bash
Nexus 5X (Android 9.+) device emulator
Dev/debugging mode
"react-native": "0.59.8"
"@storybook/addon-backgrounds": "^5.2.0-beta.17"
"@storybook/addon-knobs": "^5.2.0-beta.17"
"@storybook/addon-ondevice-backgrounds": "^5.2.0-beta.17"
"@storybook/addon-ondevice-knobs": "^5.2.0-beta.17"
"@storybook/addon-ondevice-notes": "^5.2.0-beta.17"
"@storybook/react-native": "^5.2.0-beta.17"

System:
    OS: macOS 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
Binaries:
    Node: 8.11.2 - ~/.nvm/versions/node/v8.11.2/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.2/bin/npm
Browsers:
    Chrome: 75.0.3770.100
    Safari: 12.1.2
```

**Additional context**
No issues pre-migration on `4.1.6`
Reproducible on `5.1.9` and `5.20-beta-17`

**Comment by ChuanxuMa** [comment]
meets the same error

**Comment by ChuanxuMa** [comment]
It's because your default component on initial has no any notes. To avoid it, suggest to add note on every component.

**Comment by perpetuatingcuriosity** [comment]
Even if I add notes to every story I get `TypeError: Cannot read property of 'selection' of null`

```
    "@storybook/addon-ondevice-notes": "^5.1.11",
    "@storybook/addon-storyshots": "^5.1.11",
    "@storybook/addons": "^5.1.11",
    "@storybook/react-native": "^5.1.11",
    "@storybook/react-native-server": "^5.1.11",
    "react-native": "0.61.0-rc.2",
```

Can the workaround label be removed here?

**Comment by ghondar** [comment]
in `addon-ondevice-notes/src/register.js` they have: 

```Javascript
...
const { active, api } = this.props;

    if (!active) {
      return null;
    }

    const story = api
      .store()
      .getStoryAndParameters(this.state.selection.kind, this.state.selection.story);
    const text = story.parameters[PARAM_KEY];
...
```

where `active` prop validates the getStoryAndParameters operation, but active props first time is true and the state isn't created, I added this line below if conditional

```Javascript
...
if (!active) {
   return null;
}

if (!this.state) {
   return null;
}
```

to make sure it comes with data before rendering the view.

**Comment by ghondar** [comment]
here my pr storybookjs/storybook#8152

**Comment by Gongreg** [comment]
This issue is already fixed in 5.3.0-alpha.37 . Closing it.

**Comment by Gongreg** [comment]
I am taking it back, the issue still persists. Creating a PR.

**Comment by shilman** [comment]
Jeepers creepers!! I just released https://github.com/storybookjs/storybook/releases/tag/v5.3.0-alpha.39 containing PR storybookjs/storybook#8692 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.

**Comment by leotm** [comment]
jus curious how to fix on preview screen on 5.3

```json
"@storybook/react-native": "5.3.25",
"@storybook/addon-notes": "5.3.21",
"@storybook/addon-ondevice-notes": "5.3.23",
```

https://user-images.githubusercontent.com/1881059/130494019-136510d3-0f8f-4c11-a738-822a43f59d53.mp4

the rest work well (besides need to switch tab to update)

https://user-images.githubusercontent.com/1881059/130494260-24bf24d4-c51a-484c-9248-b96a3cedbf82.mp4

(repro [repo](https://github.com/leotm/react-native-template-typescript))
